base/abstractarray.jl

```diff
@@ -839,13 +860,13 @@ function hvcat{T}(rows::Tuple{Vararg{Int}}, as::AbstractMatrix{T}...)
                 throw(ArgumentError("mismatched height in block row $(i)"))
             end
             if c-1+szj > nc
-                throw(ArgumentError("block row $(i) has mismatched number of columns"))
+                throw(ArgumentError("block row $(i) has mismatched number of columns: $(c-1+szj) $nc"))
             end
             out[r:r-1+szi, c:c-1+szj] = Aj
             c += szj
         end
         if c != nc+1
-            throw(ArgumentError("block row $(i) has mismatched number of columns"))
+            throw(ArgumentError("block row $(i) has mismatched number of columns $(c-1) $nc"))
```

Comment:
Same here and below: just printing the sizes without any punctuation or explanation is not very clear.